bug 1547396 fix disabling self-provisioning steps

kalexand-rh opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1547396

These commands are different in 3.7. The PR for 3.7 is here: https://github.com/openshift/openshift-docs/pull/9324

This bug also applies to branches before 3.6, but the commands are different. The PR for earlier branches is here: https://github.com/openshift/openshift-docs/pull/9276

mburke5678 commented on admin_guide/managing_projects.adoc:
```diff
@@ -90,39 +90,69 @@ to all authenticated developers by default.
+privileges.
+. Remove the `self-provisioners`
+xref:../architecture/additional_concepts/authorization.adoc#roles[cluster role]
+from the group.
```
Remove from what group? Or, is this obvious?

kalexand-rh replied:
I read it as `system:authenticated` as the group. @xltian, is that right?

kalexand-rh replied:
@barleyer, is `system:authenticated` the group here?

barleyer replied:
@kalexand-rh ，according to `self-provisioners` clusterrolebinding file，it should be `system:authenticated:oauth`.

mburke5678 commented on admin_guide/managing_projects.adoc:
```diff
@@ -90,39 +90,69 @@ to all authenticated developers by default.
+project. This parameter value is a string that will be presented to a user in
+the web console and command line when the user attempts to self-provision a project.
+You might use one of the following messages:
++
```
Cool!!!

mburke5678 commented on admin_guide/managing_projects.adoc:
```diff
@@ -90,39 +90,69 @@ to all authenticated developers by default.
++
+----
+$ oc patch clusterrolebinding.rbac self-provisioners -p '{ "metadata": { "annotations": { "rbac.authorization.kubernetes.io/autoupdate": "false" } } }'
+----
```
Not sure about the "by using automation" line. Makes me think of Ansible. 
How about ```

To update the role binding, edit the *self-provisioners* cluster role binding:
:...
Or,  use the following command:
```

kalexand-rh replied:
That's fair. I think it is mean to be dropped in your own automated scripts.

mburke5678 commented:
@kalexand-rh A couple of nits. Otherwise LGTM

barleyer commented on admin_guide/managing_projects.adoc:
```diff
@@ -90,39 +90,42 @@ to all authenticated developers by default.
++
 ----
-Contact your system administrator at projectname@example.com to request a project.
+$ oc adm policy remove-cluster-role-from-group self-provisioner system:authenticated system:authenticated:oauth
```
i think only `system:authenticated:oauth` is enough.

kalexand-rh commented:
@barleyer, thank you! I made a couple of changes:
- I added a step to run `oc  describe clusterrolebinding.rbac self-provisioners` to see what it's binding.
- I edited your step 2 and added it.
- I put step 4 back.

Please let me know if this looks good now. Thanks!

kalexand-rh commented:
@barleyer approved this PR in the source bug. Merging.

kalexand-rh commented:
/cherrypick enterprise-3.10

kalexand-rh commented:
/cherrypick enterprise-3.9

Xingweiyang520 commented:
hi guys, i'm using ocp 3.9.14, but i can't use the above solutions:
first, there is no self-provisioners existing.
sencond, even if i run command  " oc patch clusterrolebinding.rbac self-provisioner -p '{ "metadata": { "annotations": { "rbac.authorization.kubernetes.io/autoupdate": "false" } } }'  " , it still can create project for normal users.
